core/amber/src/main/python/core/runnables/main_loop.py

```diff
@@ -308,6 +308,10 @@ def _process_end_of_output_ports(self, _: EndOfOutputPorts) -> None:
 
         :param _: EndOfOutputPorts
         """
+        # Special case for the hack of input port dependency.
```

Comment:
if possible, move this outside this function: when there is no output port (the special hacky case), do not call process end of output ports.